Documentation/AdvancedUsage.md

```diff
@@ -520,6 +521,83 @@ open func retry(_ request: Request, for session: Session, dueTo error: Error, co
 }
 ```
 
+### `AuthenticationInterceptor`
+Alamofire's `AuthenticationInterceptor` class is a `RequestInterceptor` designed to handle the queueing and threading complexity involved with authenticating requests.
+It leverages an injected `Authenticator` protocol that manages the lifecycle of the matching `AuthenticationCredential`.
+Here is a simple example of how an `OAuthAuthenticator` class could be implemented along with an `OAuthCredential`.
+
+**`OAuthCredential`**
+
+```swift
+struct OAuthCredential: AuthenticationCredential {
+    let accessToken: String
+    let refreshToken: String
+    let userID: String
+    let expiration: Date
+
+    // Require refresh if within 5 minutes of expiration
+    var requiresRefresh: Bool { Date(timeIntervalSinceNow: 60 * 5) > expiration }
```

Comment:
I don't believe this is how `Date` works. `Date` is generated in UTC and is independent of calendar and time zone. Therefore, a date generated in Portland at the same time in Detroit would have the exact same underlying value. However, when you print them they will display in your local time zone and appear three hours apart.

In regards to this specific example, a `Date` generated pre-DST compared against a `Date` generated post-DST wouldn't matter and would work as expected because each `Date` doesn't have the concept of DST. They are simply points in time irrespective of calendar and time zone, so the comparison should be valid.

Reply:
You're right. If you have an expiration `Date`, a fixed calculation should give you the proper time interval.